sdk/ai/ai-inference/review/ai-inference-rest.api.md

```diff
@@ -0,0 +1,449 @@
+    model: string;
+    usage: CompletionsUsageOutput;
+}
+
+// @public
+export type ChatCompletionsResponseFormat = string | "text" | "json_object";
+
+// @public
+export type ChatCompletionsToolCall = ChatCompletionsToolCallParent | ChatCompletionsFunctionToolCall;
+
+// @public
+export type ChatCompletionsToolCallOutput = ChatCompletionsToolCallOutputParent | ChatCompletionsFunctionToolCallOutput;
+
+// @public
+export interface ChatCompletionsToolCallOutputParent {
+    id: string;
+    // (undocumented)
+    type: string;
+}
+
+// @public
+export interface ChatCompletionsToolCallParent {
+    id: string;
+    // (undocumented)
+    type: string;
+}
+
+// @public
+export type ChatCompletionsToolDefinition = ChatCompletionsToolDefinitionParent | ChatCompletionsFunctionToolDefinition;
+
+// @public
+export interface ChatCompletionsToolDefinitionParent {
+    // (undocumented)
+    type: string;
+}
+
+// @public
+export type ChatCompletionsToolSelectionPreset = string | "auto" | "none";
+
+// @public
+export interface ChatRequestAssistantMessage extends ChatRequestMessageParent {
+    content?: string | null;
+    role: "assistant";
+    tool_calls?: Array<ChatCompletionsToolCall>;
+}
+
+// @public
+export type ChatRequestMessage = ChatRequestMessageParent | ChatRequestSystemMessage | ChatRequestUserMessage | ChatRequestAssistantMessage | ChatRequestToolMessage;
+
+// @public
+export interface ChatRequestMessageParent {
+    // (undocumented)
+    role: ChatRole;
+}
+
+// @public
+export interface ChatRequestSystemMessage extends ChatRequestMessageParent {
+    content: string;
+    role: "system";
+}
+
+// @public
+export interface ChatRequestToolMessage extends ChatRequestMessageParent {
+    content: string | null;
+    role: "tool";
+    tool_call_id: string;
+}
+
+// @public
+export interface ChatRequestUserMessage extends ChatRequestMessageParent {
+    content: string;
+    role: "user";
+}
+
+// @public
+export interface ChatResponseMessageOutput {
+    content: string | null;
+    role: ChatRoleOutput;
+    tool_calls?: Array<ChatCompletionsToolCallOutput>;
+}
+
+// @public
+export type ChatRole = string | "system" | "user" | "assistant" | "tool";
+
+// @public
+export type ChatRoleOutput = string | "system" | "user" | "assistant" | "tool";
+
+// @public
+export type CompletionsFinishReasonOutput = string | "stop" | "length" | "content_filter" | "tool_calls";
+
+// @public
+export interface CompletionsUsageOutput {
+    capacity_type: CapacityTypeOutput;
+    completion_tokens: number;
+    prompt_tokens: number;
+    total_tokens: number;
+}
+
+// @public
+function createClient(endpointParam: string, credentials: TokenCredential | KeyCredential, options?: ClientOptions): ModelClientContext;
+export default createClient;
+
+// @public
+export type EmbeddingEncodingFormat = string | "base64" | "binary" | "float" | "int8" | "ubinary" | "uint8";
+
+// @public
+export interface EmbeddingInput {
+    image: string;
+    text?: string;
+}
```

Comment:
I see this being the same between /embeddings and /images/embeddings, is that intentional? I thought /embeddings would take text only as input and /images/embeddings would take image only as input. Also, why is text optional and image required?

Reply:
removing embeddings apis and types from v1